valohai_yaml/pipelines/conversion.py

```diff
@@ -60,8 +93,15 @@ def convert_executionlike_node(self, node: Union[ExecutionNode, TaskNode]) -> Co
         if not step:  # pragma: no cover
             raise ValueError(f"Step {step_name} not found in {self.config}")
         step_data = step.serialize()
-        step_data.update(override)
 
+        if node_data.get("mode"):
+            inputs = override.pop("inputs", {})
+            inputs = [OrderedDict(input) for input in inputs] if inputs else []
+            merged_inputs = self._merge_overridden_inputs(inputs, step_data.get("inputs", []))
+            #todo: this is a hack to get the extend inputs but it needs to be extended to other fields as well
```

Comment:
So... this only works for `inputs` right now? (I mean, changing `parameters` defaults is also a very valid use case.) That should be probably made clearer somehow.

In fact, maybe for future proofing, we should already make `override_mode` a mapping á la

```yaml
override-mode:
  inputs: override
  parameters: merge
```
– WDYT? Too much complexity?

Reply:
can do, but is it really a use case? it should works for all variables there. I think it is too complex.